### ISSUE TITLE
sklearn's GraphLassoCV or ConnectivityMeasure(cov_estimator=GraphLassoCV) ?

### ISSUE THREAD

**Issue by salma1601** [nilearn/nilearn#1213]
Many examples fit  `sklearn.covariance.GraphLassoCV`. We can rely on `ConnectivityMeasure` instead. One drawback is that we will not be able to assess the `GraphLasso` attributes, such as the `alpha_`.

**Comment by bthirion** [comment]
There is only one example playing with alpha. It can be changed in the others.

**Comment by GaelVaroquaux** [comment]
> Many examples fit sklearn.covariance.GraphLassoCV. We can rely on
> ConnectivityMeasure instead. One drawback is that we will not be able to assess
> the GraphLasso attributes, such as the alpha_.

In the example that plays with alpha_, how about we add
ConnectivityMeasure in a section after that.

We should start using more sections in examples. They make for better
readability.

**Comment by GaelVaroquaux** [comment]
I thought about this in the context of the rationalization of the examples. The way that I see it is that we should introduce and advertise ConnectivityMeasure for group-level / multi subject analysis, but not when we are discussing single-subject analysis. It brings complexity but little benefits.

**Comment by salma1601** [comment]
>It brings complexity but little benefits.

I am hesitating. So you think it's preferable to teach how covariance estimators work for 1 subject, then say these estimators are encompassed in ConnectivityMeasure which is suitable for a group. 

May be it's the right way, but I stick to normalizing the covariances/precisions. Plus in all examples we standardize, so  actually we have correlations but are calling them covariances. All this is mixing up things for someone who is thinking in terms of connections.

**Comment by GaelVaroquaux** [comment]
> So you think it's preferable to teach how covariance estimators work
> for 1 subject, then say these estimators are encompassed in
> ConnectivityMeasure which is suitable for a group.

Well, we still have to import them, and so talk about them. The problem
with ConnectivityMeasure with one subject is that it forces this tricky
thing of using a list to pass the data in ConnectivityMeasure, and to
take the first element of that list out. It's easy to get wrong, and not
get an very useful error message.

> May be it's the right way, but I stick to normalizing the covariances/
> precisions. Plus in all examples we standardize, so actually we have
> correlations but are calling them covariances. All this is mixing up
> things for someone who is thinking in terms of connections.

Possibly. But I would think that these people need to do multiple
subjects, or multiple sessions, in which case, they should be using
ConnectivityMeasure anyhow.

**Comment by AlexandreAbraham** [comment]
> The problem with ConnectivityMeasure with one subject is that it forces this tricky thing of using a list to pass the data

This problem is recurrent in a lot of functions in nilearn. Why not try to tackle it once and for all?

One option that I find elegant (I can bet that some people will not like it) could be:
```python

class single(object):
    def __init__(self, imgs):
        self.imgs = imgs

def allow_single(func):
    def wrapper(*args, **kwargs):
        if len(args) > 0 and isinstance(args[0], single):
            res = func([args[0].imgs], *args[1:], **kwargs)[0]
        else
            res = func(*args, **kwargs)
        return res
    return wrapper

@allow_single
def multi_function(list_of_imgs):
    pass
```

Usage:

```python
multi_function(group_data)
multi_function(single(subject_data))
```

**Comment by bthirion** [comment]
GraphLassoCV is used in 4 examples. This should be improved indeed.
